boilerplate/openshift/golang-osd-operator/update

```diff
@@ -10,14 +10,45 @@ source $CONVENTION_ROOT/_lib/common.sh
 # Expect POST
 [[ "$1" == "POST" ]] || err "Got a parameter I don't understand: '$1'. Did the infrastructure change?"
 
+# Add codecov configuration
 echo "Copying .codecov.yml to your repository root."
 cp ${HERE}/.codecov.yml $REPO_ROOT
 
-# TODO: boilerplate more of Dockerfile
+# Add dependabot configuration
+mkdir -p $REPO_ROOT/.github
+if [ ! -f ${REPO_ROOT}/.github/dependabot.yml ]; then
+    # Only copy dependabot config if it doesn't exist (to allow for custom configurations)
+    echo "Copying dependabot.yml to .github/dependabot.yml"
+    cp ${HERE}/dependabot.yml ${REPO_ROOT}/.github/dependabot.yml
+fi
+
+# Update Dockerfile builder image
 DOCKERFILE=build/Dockerfile
 echo "Overwriting $DOCKERFILE's initial FROM with $IMAGE_PULL_PATH"
 ${SED?} -i "1s,.*,FROM $IMAGE_PULL_PATH AS builder," $DOCKERFILE
 
+# Determine latest ubi standard, minimal, and micro image tags and update Docker runtime image
+UBI_STANDARD_IMAGE_LATEST=registry.access.redhat.com/ubi8/ubi:latest
+UBI_MINIMAL_IMAGE_LATEST=registry.access.redhat.com/ubi8/ubi-minimal:latest
+UBI_MICRO_IMAGE_LATEST=registry.access.redhat.com/ubi8/ubi-micro:latest
+UBI_STANDARD_IMAGE_RELEASE=$(skopeo inspect --override-os linux --override-arch amd64 docker://${UBI_STANDARD_IMAGE_LATEST} --format "{{.Name}}:{{.Labels.version}}-{{.Labels.release}}")
+UBI_MINIMAL_IMAGE_RELEASE=$(skopeo inspect --override-os linux --override-arch amd64 docker://${UBI_MINIMAL_IMAGE_LATEST} --format "{{.Name}}:{{.Labels.version}}-{{.Labels.release}}")
+UBI_MICRO_IMAGE_RELEASE=$(skopeo inspect --override-os linux --override-arch amd64 docker://${UBI_MICRO_IMAGE_LATEST} --format "{{.Name}}:{{.Labels.version}}-{{.Labels.release}}")
+
+if grep -q ${UBI_STANDARD_IMAGE_LATEST} $DOCKERFILE; then
+    echo "Overwriting ${DOCKERFILE}'s ${UBI_STANDARD_IMAGE_LATEST} image to ${UBI_STANDARD_IMAGE_RELEASE}"
+    ${SED?} -i "s,${UBI_STANDARD_IMAGE_LATEST},${UBI_STANDARD_IMAGE_RELEASE}," $DOCKERFILE
+elif grep -q ${UBI_MINIMAL_IMAGE_LATEST} $DOCKERFILE; then
+    echo "Overwriting ${DOCKERFILE}'s ${UBI_MINIMAL_IMAGE_LATEST} image to ${UBI_MINIMAL_IMAGE_RELEASE}"
+    ${SED?} -i "s,${UBI_MINIMAL_IMAGE_LATEST},${UBI_MINIMAL_IMAGE_RELEASE}," $DOCKERFILE
+elif grep -q ${UBI_MICRO_IMAGE_LATEST} $DOCKERFILE; then
+    echo "Overwriting ${DOCKERFILE}'s ${UBI_MICRO_IMAGE_LATEST} image to ${UBI_MICRO_IMAGE_RELEASE}"
+    ${SED?} -i "s,${UBI_MICRO_IMAGE_LATEST},${UBI_MICRO_IMAGE_RELEASE}," $DOCKERFILE
+else
+    echo "Did not detect any ubi latest tags in ${DOCKERFILE}, skipping replacement."
+fi
```

Comment:
Suggested improvement. This dedupes some of the logic, as well as only makes the network calls if we detect a latest tag and only for the specific image in question.

```suggestion
# Update any UBI images using latest tags to use a versioned tag that is compatible with dependabot
for ubi-latest in $(grep -oE 'registry.access.redhat.com/ubi[7-9]/ubi.*?:latest' ${DOCKERFILE}); do
    replacement_image=$(skopeo inspect --override-os linux --override-arch amd64 docker://${ubi-latest} --format "{{.Name}}:{{.Labels.version}}-{{.Labels.release}}")
    echo "Overwriting ${DOCKERFILE}'s ${ubi-latest} image to ${replacement_image}"
    ${SED?} -i "s,${ubi-latest},${replacement_image}," ${DOCKERFILE}
done
```

Reply:
Much better! There were some issues with your suggestion I fixed after local testing.